### PR TITLE
CI workflow

### LINKED ISSUE #1: Enable CI
Examples used to be tested in the Lingua Franca JUnit-based test framework. Now that they have moved, the examples should be tested separately.

### PR DESCRIPTION
**Pull request by lhstrh**
This PR enables CI for this repo, which involves traversing all files and checking for each `.lf` file whether it compiles.

Fixes #1.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+jobs:
+  check-lf-files:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Java 17
+        run: |
+          echo "$JAVA_HOME_17_X64/bin" >> $GITHUB_PATH
+          echo "org.gradle.java.home=${JAVA_HOME_17_X64//\\/\/}" >> gradle.properties
```
Comment (lhstrh) [inline .github/workflows/ci.yml:17]: I copied this from elsewhere, but in hindsight, this doesn't seem right to me. At the cwd, there is no `gradle.properties`...

Comment (lhstrh) [inline .github/workflows/ci.yml:17]:
```suggestion
```

**Comment by petervdonovan** [comment]
Some of the errors are due to the refactoring that I did this spring. They are not bugs in LFC, exactly, just things about the examples that need to be brought up to date.

**Comment by lhstrh** [comment]
> Some of the errors are due to the refactoring that I did this spring. They are not bugs in LFC, exactly, just things about the examples that need to be brought up to date.

The bugs that I'm talking about are ones like these:
```
Error: Command failed: lfc  "./C/src/DistributedHelloWorld/HelloWorldDecentralizedSTP.lf"
Exception in thread "pool-1-thread-2" java.lang.NullPointerException: Cannot invoke "java.util.LinkedHashSet.stream()" because the return value of "org.lflang.ast.ASTUtils.allFileLevelPreambles(org.lflang.lf.Reactor)" is null
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (lhstrh) [reply .github/workflows/ci.yml]: Looks like we don't need it...

**Comment by lhstrh** [comment]
I've gone ahead and merged this; lets create a separate PR to address the failures.

**Pull request merged**